plugins/qcheck-stm/src/stm_of_ir.ml

```diff
@@ -143,6 +143,67 @@ let ty_show_of_core_type inst ty =
 
 let exp_of_ident id = pexp_ident (lident (str_of_ident id))
 
+(** This function should be specialized `flatten_arrow (Cfg.is_sut config)`
+    [when_sut] returns an [option] in order to notify the function whether we
+    keep the value or not *)
+let flatten_arrow is_sut when_sut when_not core_type =
```

Comment:
As it stands, this function is used only once and not using all the genericity it comes with. Do we envision its full use?

Reply:
We get rid of it.